### ISSUE TITLE
/vendor/etc/audio/audio_policy_configuration.xml not handled properly

### ISSUE THREAD

**Issue by cawilliamson** [Zackptg5/Audio-Modification-Library#1]
Hi there!

I'm using TreskMod (ROM based on OmniRom) and as you can see below it copies different audio_policy_configuration.xml files to /vendor/etc/audio and /vendor/etc/:

https://github.com/omnirom/android_hardware_qcom_audio-caf-msm8998/blob/android-8.1/configs/sdm845/sdm845.mk#L94

It looks like the file from /vendor/etc/ is used when in fact the one from /vendor/etc/audio/ should be used. The one in /vendor/etc/ ends up in a broken state where in-call audio is on speakerphone only instead of the earpiece.

I have managed to adapt my mod to overcome this in a super hacky way by copying:

/vendor/etc/audio/audio_policy_configuration.xml -> /vendor/etc/audio_policy_configuration.xml

Obviously since this is directly replacing a file though it's not ideal.

I'd appreciate any assistance you can give. 👍 

Thanks!

**Comment by therealahrion** [comment]
Well this isn't an AML problem but a ROM problem. But it's not really a problem at all. Omni does stuff their own way and this is one of them. Most devices use vendor/etc/audio policy stuff but op5/5t use vendor/etc/audio/audio policy (as you've seen). However, omni doesn't bother with that extra folder as theirs no need for it really so they just moved it all to the normal vendor/etc directory.

Long story short, all omni based roms are like that and typically work fine so the speakerphone issue you're having may be due to some other audio mod (I'm on a different omni based rom - omnidragon, and don't have the problems you're describing)

**Comment by Zackptg5** [comment]
Well this isn't an AML problem but a ROM problem. But it's not really a problem at all. Omni does stuff their own way and this is one of them. Most devices use vendor/etc/audio policy stuff but op5/5t use vendor/etc/audio/audio policy (as you've seen). However, omni doesn't bother with that extra folder as theirs no need for it really so they just moved it all to the normal vendor/etc directory.

Long story short, all omni based roms are like that and typically work fine so the speakerphone issue you're having may be due to some other audio mod (I'm on a different omni based rom - omnidragon, and don't have the problems you're describing)
